### PR TITLE
LPS-178022 Remove PortletPreferencesHelper from portal-search-web

### PR DESCRIPTION
**Pull request by julschong**

### CODE REVIEW

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/category/facet/portlet/CategoryFacetPortletPreferencesImpl.java [hunk 1]
```diff
@@ -30,58 +32,72 @@
...
+		return StringUtil.trim(
+			_portletPreferences.getValue(
+				CategoryFacetPortletPreferences.PREFERENCE_KEY_PARAMETER_NAME,
+				"category"));
 	}
 
 	@Override
 	public String[] getVocabularyIds() {
-		String vocabularyIds = _portletPreferencesHelper.getString(
-			CategoryFacetPortletPreferences.PREFERENCE_VOCABULARY_IDS, null);
```
Comment (julschong) [inline modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/category/facet/portlet/CategoryFacetPortletPreferencesImpl.java:73]: null or blank, StringUtil.split returns empty String array

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/search/bar/portlet/SearchBarPortletPreferencesImpl.java [hunk 2]
```diff
@@ -31,49 +33,50 @@
...
 	public SearchScopePreference getSearchScopePreference() {
-		Optional<String> valueOptional = _portletPreferencesHelper.getString(
-			SearchBarPortletPreferences.PREFERENCE_KEY_SEARCH_SCOPE);
-
-		Optional<SearchScopePreference> searchScopePreferenceOptional =
-			valueOptional.map(SearchScopePreference::getSearchScopePreference);
-
-		return searchScopePreferenceOptional.orElse(
-			SearchScopePreference.THIS_SITE);
+		return SearchScopePreference.getSearchScopePreference(
```
Comment (julschong) [inline modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/search/bar/portlet/SearchBarPortletPreferencesImpl.java:75]: if preferenceString is null or blank, defaults to THIS_SITE

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
ci:test:search

**Comment by tinatian** [comment]
@julschong I changed the solution to avoid fetching from an empty portlet preference instance, please see https://github.com/liferay-core-infra/liferay-portal/pull/4401

**Pull request closed without merging**